### PR TITLE
[coreimage] Add new headers API up to beta 6

### PR DESCRIPTION
**Pull request by spouliot**
Filters are not complete (ref: #57350)
https://bugzilla.xamarin.com/show_bug.cgi?id=57350

### CODE REVIEW

**Comment by spouliot** [comment]
missing a few constants (need to validate where the keys are used first)

File: src/coreimage.cs [hunk 1]
```diff
@@ -1692,6 +1805,13 @@ interface CIImageProcessorInput {
 		[NullAllowed, Export ("metalTexture")]
 		IMTLTexture MetalTexture { get; }
 #endif
+
+		// @required but it was added in Xcode9
```
Comment (rolfbjarne) [inline src/coreimage.cs]: Maybe add `[Abstract]` in `#if XAMCORE_4_0`?

File: src/coreimage.cs [hunk 2]
```diff
@@ -1728,6 +1848,13 @@ interface CIImageProcessorOutput {
 		[NullAllowed, Export ("metalCommandBuffer")]
 		IMTLCommandBuffer MetalCommandBuffer { get; }
 #endif
+
+		// @required but it was added in Xcode9
```
Comment (rolfbjarne) [inline src/coreimage.cs]: Maybe add `[Abstract]` in `#if XAMCORE_4_0`?

**Comment by spouliot** [comment]
note: filled bugs for remaining (minor) issues, so we can provide the API without them

**Pull request merged**